server/s3_object_data_copier.cc

```diff
@@ -349,7 +349,11 @@ void S3ObjectDataCopier::copy(
 
   motr_reader = motr_reader_factory->create_motr_reader(
       request_object, src_obj_id, layout_id, pvid, motr_api);
-  motr_reader->set_last_index(0);
+  // motr_reader->set_last_index(0);
+
+  size_t block_start_offset =
+      first_byte_offset - (first_byte_offset % motr_unit_size);
```

Comment:
need validation of 'first_byte_offset' before being used and if invalid value found, set it to 0

Reply:
Validations are done in `validate_range_header()` function in `s3_put_multiobject_copy_action.cc`

Reply:
right, realized